### ISSUE TITLE
Ideas for breaking changes

### ISSUE THREAD

**Issue by niklasf** [niklasf/python-chess#659]
Changes to consider when the opportunity arises.

* `chess.svg.board(..., flipped)` -> `chess.svg.board(..., orientation)`
* Remove `chess.pgn.BaseVisitor.parse_san()`

**Comment by PedanticHacker** [comment]
(1) Niklas, why do you want to rename the `flipped` argument name of `chess.svg.board()` to `orientation`? 🤔  If you think of renaming it, you can do it right now. Make it as a 1.x breaking change. Everyone have not upgraded to 1.x yet, so the opportunity is basically right now. Version 1.0.2 can introduce this. The changelog should explain this and people will adapt. And there's only one simple name to change, not a hassle at all.

(2) Do you want to remove `parse_san()` abstract method of the `BaseVisitor()` abstract base class altogether, or just remove `return board.parse_san(san)` and make it `pass` instead?

**Comment by PedanticHacker** [comment]
Here's an idea of a breaking change: since the introduction of passing a dictionary as the `colors` argument to `chess.svg.board()`, we can now remove the `style` argument in there. Thoughts?

**Comment by niklasf** [comment]
Good point about `style`. And no, I will not sneak breaking changes into minor releases. This is just a list of things to consider in the future.

**Comment by PedanticHacker** [comment]
A little off-topic here, but python-chess v1.1.0 is still not available via pip to upgrade, and the latest version for python-chess on PyPI shows as 1.0.1. Have you not uploaded the new version yet?

**Comment by niklasf** [comment]
* Remove `chess.engine.PovWdl` bc for `Tuple[int, int, int]`.

**Comment by PedanticHacker** [comment]
- Remove `flipped` and `style` parameters of `chess.svg.board()`.

**Comment by niklasf** [comment]
* Distinguish PGN root and child nodes for typing

**Comment by khink** [comment]
Not sure if i should post here, but here goes:

I'm looking at what is needed to create an interface/engine for a 5x5 board. Am i correct in assuming this would be a major, breaking change in this module? Looking at the [constants](chess/__init.py) like SQUARES, FILE_NAMES, RANK_NAMES, and also the various [variants](chess/variants.py), this is not easily changed in the current situation, right?

The reason i'm asking is that i saw the [mini chess app](https://apps.apple.com/us/app/mini-chess-on-watch/id1480844881) which uses a 5x5 board. I was hoping to play it on my computer in some way. Having looked at [a YouTube video](https://www.youtube.com/watch?v=U4ogK0MIzqk) where someone coded a chess engine+interface from the ground up, i was inspired but also realized that there are many projects out there, and i was hoping to re-use / contribute whenever possible.

**Comment by niklasf** [comment]
Hi, I tried mini chess a while back and it's definitely fun. For this particular library I would consider it out of scope, with so many places using bitboards with 64 squares it would complicate everything significantly.

**Comment by niklasf** [comment]
* `Status.IMPOSSIBLE_MATERIAL`, merging the various individual reasons but also more pedantically counting promoted pieces.

**Comment by niklasf** [comment]
* More flexible SVG rendering. Composable layers?

**Comment by niklasf** [comment]
* Remove `max_fds` from Syzygy. File descriptors can be closed after creating memory maps.

**Comment by niklasf** [comment]
* Positional only arguments
